### PR TITLE
indexer-envio: positive MARKET_HOURS probe + transceiver-filtered fallback drain

### PR DESCRIPTION
**Pull request by chapati23**
## Summary

Three BACKLOG items from `Envio v3 Migration Follow-Ups` + `Indexer-envio defensive-hardening follow-ups (from rpc + bridge PR)`.

1. **Positive MARKET_HOURS identification in `fetchBreakerKind`.** Adds an `isFXMarketOpen(uint256)` selector probe after the MD/VD probes. When all three positive probes miss, returns `null` instead of defaulting to `MARKET_HOURS`. `breakerKindEffect` already skips its cache on `null`, so the next event re-probes rather than persisting a misclassification (catches future breaker kinds, proxy-upgraded breakers, EOAs added via governance / RPC poisoning). Warn signature renamed: `breakers.fetchBreakerKind.market_hours_default` → `breakers.fetchBreakerKind.unknown_kind`. Grep confirmed no alert rule keys on the old prefix.

2. **Transceiver filter on the Wormhole `TransferRedeemed` fallback drain.** When `MessageAttestedTo` doesn't fire before `TransferRedeemed` (HyperSync drops the attest log, historical backfill, replay), the fallback drain now passes `mgr?.transceiver` from the NTT manifest to the existing `drainDestPending` predicate. In a multi-NTT-inbound same-tx the wrong-transceiver scratch rows are skipped and source identity stays unstamped rather than mis-attributed. Manifest miss falls through unfiltered (same as today, recoverable via `pnpm generate:ntt-addresses` + replay).

3. **Envio v3 backfill validation closed.** `envio-cloud indexer get` shows the indexer is on `tier=medium` and the latest deploy reached `timestamp_caught_up_to_head_or_endblock` in ~38 min from registration — same 15-40 min range as the v2 baseline despite the meaningfully larger event surface (multichain Celo+Monad, Liquity v2 CDPs, Wormhole bridge handlers). Medium-tier upgrade was kept, not deferred. Removes the BACKLOG entry; updates the envio Skill (`small` → `medium`).

Also fixes a cross-validated test-mock semantic discovered during /review + /codex-review: `_setMockBreakerKind(_, _, null)` now propagates `null` instead of coercing to `MARKET_HOURS`, matching the new "null = inconclusive" contract.

## Test plan

- [x] `pnpm --filter @mento-protocol/indexer-envio typecheck` — clean (with `exactOptionalPropertyTypes` from #604)
- [x] `pnpm --filter @mento-protocol/indexer-envio test` — 770 passing (49 files); 2 new bridge tests + 2 updated breaker tests
- [x] `pnpm --filter @mento-protocol/indexer-envio lint` — clean (baseline absorbed)
- [x] `trunk fmt` + `trunk check` — clean
- [x] `pnpm exec turbo run lint --filter=@mento-protocol/indexer-envio` — clean (re-checked after comment compression to absorb the eslint-baseline-diff 30-line shift threshold)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Changes breaker RPC classification and Wormhole source attribution paths used in production indexing; behavior is safer but affects how unknown breakers and multi-NTT txs are stored until re-sync.
> 
> **Overview**
> Hardens **indexer-envio** RPC classification and Wormhole bridge attribution, and closes related backlog/docs.
> 
> **Breaker kind probing** no longer defaults unknown contracts to `MARKET_HOURS`. After MedianDelta and ValueDelta selector checks, it adds a positive **`isFXMarketOpen`** probe (`MARKET_HOURS_BREAKER_ABI`). If nothing matches, `fetchBreakerKind` returns **`null`**, logs `breakers.fetchBreakerKind.unknown_kind`, and **`breakerKindEffect`** skips caching so the next event re-probes. Test HTTP mocks and unit tests follow the new contract (`null` mock no longer coerces to `MARKET_HOURS`).
> 
> **Wormhole `TransferRedeemed` fallback drain** (when `MessageAttestedTo` is missing) now passes **`mgr?.transceiver`** from the NTT manifest into `drainDestPending`, so multi-NTT same-tx replays skip wrong-transceiver scratch rows instead of mis-stamping source identity; manifest miss keeps the prior unfiltered degraded path. New handler tests cover filtered vs manifest-miss behavior.
> 
> **Housekeeping:** removes completed Envio v3 / defensive-hardening items from `BACKLOG.md`; Envio skill docs now say the mento indexer is on the **`medium`** tier (was `small`).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8d93b98d04473387c1478bd1ecb589e88479b43f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: indexer-envio/src/rpc/breakers.ts [hunk 1]
```diff
@@ -269,22 +273,29 @@ export async function fetchBreakerKind(
...
+  const mhProbe = await probeFunction(
+    chainId,
+    breakerAddress,
+    MARKET_HOURS_BREAKER_ABI,
+    "isFXMarketOpen",
+    [0n],
+    log,
+  );
+  if (mhProbe === "rpc_error") return null;
+  if (mhProbe === "present") return "MARKET_HOURS";
```
[the remark this answers is not in the document]
Comment (chapati23) [reply indexer-envio/src/rpc/breakers.ts:289]: Fixed in 8d93b98d — `registerMockBreakerKindHttp` now sets the isFXMarketOpen probe mock for `MARKET_HOURS` (present, returns `true`) and for `null` (missing, raw `0x`). The bridge's three-probe shape now matches the runtime `fetchBreakerKind` contract.

**Comment by chapati23** [comment]
Closing as superseded by #606 (merged earlier today).

Discovered during conflict resolution that #606's headline change — fixing `probeFunction` to treat bare "reverted." as "missing" under viem 2.x — is load-bearing. The original `"returned no data"`-only heuristic was dead code against real Celo breakers, which means this PR's positive `isFXMarketOpen` probe is moot (MD probe returns rpc_error → `fetchBreakerKind` returns null before the new MH probe runs).

#606's cache-opt-out on MARKET_HOURS results achieves the same goal (no stuck misclassification across restarts) with a smaller, cleaner footprint.

Worth salvaging from this branch: the Envio v3 Migration Follow-Ups BACKLOG entry removal (validated against current production sync) and the envio Skill tier doc fix (`small` → `medium`). Ship those as a tiny follow-up.
